### PR TITLE
Featureswagger

### PR DESCRIPTION
**Pull request by ankitrev**

### CODE REVIEW

File: swagger/validate.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	// remove the "/api" from the schemaURKPath before matching with the requestURLPath
+	//schemaURLPath = strings.SplitAfterN(schemaURLPath, "api", 2)[1]
+	return schemaURLPath == requestURLPath
+}
+
+//isSameParametricURLPath checks if the schema's parametric path matches with the request URL path
+func isSameParametricURLPath(schemaURLPath, requestURLPath string) bool {
+	// remove the "/api" from the schemaURLPath before matching with the requestURLPath
+	if !strings.Contains(schemaURLPath, "ops") && !strings.Contains(schemaURLPath, "plugin") &&
+		!strings.Contains(schemaURLPath, "upi") {
```
Comment (udsharma015) [inline swagger/validate.go]: This shud not be a part of a generic library. No Hardcoding shud be done. 
If we want to exclude any internal APIs for validation, we can take a list of Url Pattern as an optional parameter. 

@happay  to confirm

Comment (happay) [reply swagger/validate.go]: Agree with @udsharma015

Comment (ankitrev) [reply swagger/validate.go]: @udsharma015  Okay i'll make the changes.

File: idempotency/idempotency.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const (
+	AppID      = "App-ID"
+	RequestID  = "Request-ID"
+)
+
+type BaseModel struct {
+	CreatedAt time.Time  `json:"created_at"`
+	UpdatedAt time.Time  `json:"updated_at"`
+	DeletedAt *time.Time `sql:"index" json:"deleted_at,omitempty"`
+}
```
Comment (udsharma015) [inline idempotency/idempotency.go:20]: @ankitrev  There is a basemodel already present at PropertyMap, try to use that.

**Comment by udsharma015** [comment]
@ankitrev  You may decline this PR. not Required.

**Pull request closed without merging**